File: client/lib/models/appointment.dart
import 'package:intl/intl.dart';
import 'package:nd_telemedicine/models/health_status.dart';

class Appointment {
  late int id;

  late int patient;

  late int doctor;

  late String date;

  late String time;

  late String today;

  late HealthStatus healthStatus;

  late String dateString;


  setDetails(var details){

    id = details['id'];
    patient = details['patient'];
    doctor = details['doctor'];
    date = details['date'];
    time = details ['time'];
    today = details['today'];
    if (details['dateString'] != null){
      dateString = details['dateString'];
    }


  }
}

File: server/appointment-service/src/main/java/com/lukemacdonald/appointmentservice/model/Appointment.java
package com.lukemacdonald.appointmentservice.model;


import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.util.Date;


@Entity
@Getter
@Setter
@Table(name = "appointments")
public class Appointment{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @Column
    private int  patient;

    @Column
    private String email;

    @Column
    private int doctor;

    @Column
    @Temporal(TemporalType.DATE)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "dd/MM/yyyy")
    private Date date;
    @Column
    private String time;
    @Column
    private String today;

    private String dateString;
}
